To: vim_dev@googlegroups.com
Subject: Patch 7.3.322
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.322
Problem:    #ifdef for PDP_RETVAL doesn't work, INT_PTR can be a typedef.
Solution:   Check the MSC version and 64 bit flags. (Sergiu Dotenco)
Files:	    src/os_mswin.c


*** ../vim-7.3.321/src/os_mswin.c	2011-08-10 17:07:56.000000000 +0200
--- src/os_mswin.c	2011-09-21 20:09:05.000000000 +0200
***************
*** 1781,1787 ****
  }
  
  /* Attempt to make this work for old and new compilers */
! #if !defined(_MSC_VER) || (_MSC_VER < 1300) || !defined(INT_PTR)
  # define PDP_RETVAL BOOL
  #else
  # define PDP_RETVAL INT_PTR
--- 1781,1787 ----
  }
  
  /* Attempt to make this work for old and new compilers */
! #if !defined(_WIN64) && (!defined(_MSC_VER) || _MSC_VER < 1300)
  # define PDP_RETVAL BOOL
  #else
  # define PDP_RETVAL INT_PTR
*** ../vim-7.3.321/src/version.c	2011-09-21 19:50:00.000000000 +0200
--- src/version.c	2011-09-21 20:08:29.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     322,
  /**/

-- 
Don't Panic!
		-- The Hitchhiker's Guide to the Galaxy

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
